Update to node 14.17.5, go 16.7

jwhitlock opened the pull request:
Update language versions. I didn't find documentation for updating these, so I modeled my changes after other versions:

* node 14.16.1 to 14.17.5 (modeled on PR #4734, replaces #4880)
* go 1.16.3 to 1.16.7 (modeled on PR #4743)

Most of the go changes were from a `yarn generate` run. I added a darwin arm64 go package, and manually updated the SHA-256 hashes (running `shasum` locally, and they match the [published hashes](https://golang.org/dl/))
 
 I have not build or uploaded new Docker images yet. node 14.17.5 is too fresh, and the base image needs to be added to [docker hub](https://hub.docker.com/_/node) first.

jwhitlock commented on workers/generic-worker/gw-decision-task/tasks.yml:
```diff
@@ -318,31 +318,34 @@ Mounts:
-          url: 'https://storage.googleapis.com/golang/go1.16.3.windows-amd64.zip'
-          sha256: 'a4400345135b36cb7942e52bbaf978b66814738b855eeff8de879a09fd99de7f'
+          url: 'https://storage.googleapis.com/golang/go1.16.7.windows-amd64.zip'
+          sha256: '56b3a9024268f226f679c3a8ffb21f4214a75f84050b2ha395b362ae2cc8e53e9'
```
Oops, there is an `h` that needs to be removed.

I've got this fix ready, but I'm waiting for the upstream node images.

petemoore commented:
The failures should hopefully disappear once the upstream images are available, and then the new docker images have been published. It might be best to set this back to draft until then, and rerun the CI once the images are in place. Thanks John!

jwhitlock commented:
I needed a few more changes to build and publish the updated docker images.

I added ``-platform linux/amd64`` to ``docker build`` so that it wouldn't build ``darwin/arm64`` binaries. If it is useful, we could expand this and start [building multi-platform docker manifests](https://www.docker.com/blog/multi-platform-docker-builds/).

The `worker-ci` image build was broken for me, and required a few changes. First, I added the [gpg key](https://download.docker.com/linux/ubuntu/) for Docker's apt server for Ubuntu. Previously, we fetched it from https://sks-keyservers.net/, which has shutdown the pool service. I tried switching to keys.openpgp.org first, but the key isn't there. This felt a little more honest then uploading the key so we could use a server, and I think is just as secure as the previous version.

Second, I got the error "Maximum call stack size exceeded" when installing `babel-cli`. This could have been due to my cross-compilation. A few people had this issue, but no one seemed to have a great solution. I found that I was able to install it with `yarn global add babel-cli` instead.

There are now updated images for:
* [taskcluster/browser-test:14.17.5](https://hub.docker.com/repository/docker/taskcluster/browser-test)
* [taskcluster/ci-image:node14.17.5-pg11-go1.16.7](https://hub.docker.com/repository/docker/taskcluster/ci-image)
* [taskcluster/rabbit-test:14.17.5](https://hub.docker.com/repository/docker/taskcluster/rabbit-test)
* [taskcluster/worker-ci:node14.17.5](https://hub.docker.com/repository/docker/taskcluster/worker-ci)

petemoore reviewed (approved):
Looks awesome! Many thanks for working through all the issues. 🥳
